packages/components/src/filters/advanced/number-filter.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, TextControl } from '@wordpress/components';
+import { get, find, partial } from 'lodash';
+import interpolateComponents from 'interpolate-components';
+import classnames from 'classnames';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import TextControlWithAffixes from '../../text-control-with-affixes';
+import { formatCurrency } from '../../../../currency/src';
+
+class NumberFilter extends Component {
+	getBetweenString() {
+		return _x(
+			'{{moreThan /}} and {{lessThan /}}',
+			'Numerical range inputs arranged on a single line',
+			'wc-admin'
+		);
+	}
+
+	getLegend( filter, config ) {
+		const inputType = get( config, [ 'input', 'type' ], 'number' );
+		const rule = find( config.rules, { value: filter.rule } ) || {};
+		let filterStr = filter.value || '';
+
+		if ( 'between' === rule.value ) {
+			let [ moreThan, lessThan ] = filterStr.split( ',' );
+
+			if ( 'currency' === inputType ) {
+				moreThan = formatCurrency( moreThan );
+				lessThan = formatCurrency( lessThan );
+			}
+
+			filterStr = interpolateComponents( {
+				mixedString: this.getBetweenString(),
+				components: {
+					moreThan: <Fragment>{ moreThan }</Fragment>,
+					lessThan: <Fragment>{ lessThan }</Fragment>,
+				},
+			} );
+		} else if ( 'currency' === inputType ) { // need to format a single input value as currency
```

Comment:
Do you think we could run `.split( ',' )` on all rules (not just between) and map any changes like `formatCurrency()` so we don't have to repeat them?  Or does that add more overhead overall?

Reply:
I don't think it adds any overhead, but I'm not sure it simplifies things either. If we always `.split()`, we have to figure out how to handle cases where we switch from a "between" filter that only has the first input populated (the "morethan" side) to a "less than" filter. If we split that string in the same fashion as above, `lessThan` is `undefined`.. do we use that value, or take the known good `moreThan` in its place?

Reply:
Logic got cleaned up a bit in fa6474f.